relayer/src/light_client/tendermint.rs

```diff
@@ -56,6 +63,79 @@ impl super::LightClient<CosmosSdkChain> for LightClient {
 
         self.fetch_light_block(AtHeight::At(height))
     }
+
+    /// Given a client update event that includes the header used in a client update.
+    /// it looks for misbehaviour by fetching a header at same or latest height.
+    /// TODO - return also intermediate headers.
+    fn build_misbehaviour(
+        &mut self,
+        client_state: &AnyClientState,
+        update: UpdateClient,
```

Comment:
For consistency with `verify`:

```suggestion
        update: UpdateClient,
        client_state: &AnyClientState,
```